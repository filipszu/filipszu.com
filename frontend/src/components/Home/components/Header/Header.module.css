.Logo {
  font-size: 30px;
  margin: 0px auto;
}

.Logo h1,
.Logo h4 {
  margin: 0px;
  padding: 0px;
}

.Logo h4 {
  margin-top: -10px;
  padding-left: 49px;
  color: #a2ef00;
}

@media only screen and (max-width: 1366px) {
  .Logo h4 {
    font-size: 24px;
  }
}
