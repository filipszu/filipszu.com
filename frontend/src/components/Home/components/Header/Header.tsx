import classes from "./Header.module.css";
export interface HeaderProps {
  title?: string;
  subtitle?: string;
}
export const TestID = "szu-header";
const Header = (props: HeaderProps) => {
  const title = props.title || "FiliPSZU";
  const subtitle = props.subtitle || "Software Engineer";
  return (
    <div className={classes.Logo} data-testid={TestID}>
      <h1>{title}</h1>
      <h4>{subtitle}</h4>
    </div>
  );
};
export default Header;
