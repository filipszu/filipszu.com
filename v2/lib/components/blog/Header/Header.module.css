.NavBar {
    border-bottom: 3px dashed #CB0077;
    background-color: rgba(0, 0, 0, .85);
    padding: 10px;
}

.Logo{
    position: relative;
    top: 1em;
    left: .5em;
    width: 220px;
    cursor: pointer;
    z-index: 10;
}

.Logo a {
    display: block;
}

.Logo a:hover{
    text-decoration: none;
}

.Logo h1, .Logo h4{
    color: white;
	margin: 0px;
	padding:0px;
}

.Logo h1{
    font-size: 42px;
}

.Logo h4{
    position: relative;
	top: -4px;
	left: 35px;
	color: #A2EF00;
    font-size: 24px;
}


.SiteTitle{
    text-align: center;
    position: relative;
    top: -.52em;
    font-size: 2.3em;
    color: #CB0077;
}