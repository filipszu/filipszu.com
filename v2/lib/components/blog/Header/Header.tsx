import Link from "next/link";
import classes from "./Header.module.css";
export interface HeaderProps{

}

export default function Header(props: HeaderProps){
    return (
        <nav className={classes.NavBar}>
            <div className={classes.Logo}>
                <Link href="/">
                    <div className={classes.LogoHit}>
                        <h1>Filip Szu</h1>
                        <h4>Software Engineer</h4>
                    </div>
                </Link>
            </div>
            <div className={classes.SiteTitle}>
                <Link href="/blog">Rambles From The Cut</Link>
            </div>
        </nav>
    );
}