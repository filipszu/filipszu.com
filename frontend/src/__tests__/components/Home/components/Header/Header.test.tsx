import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Header from "../../../../../components/Home/components/Header/Header";

const hardcodedTitle = "FiliPSZU";
const hardcodedSubtitle = "Software Engineer";
const mockedTitle = "Mocked Title";
const mockedSubtitle = "Mocked Subtitle";

describe(`Header component responsibble for rendering ${hardcodedTitle} ${hardcodedSubtitle}`, () => {
  test(`It will render correctly and match the snapshot.`, () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test(`[Given] no title prop [Then] it will render the hardcoded title: ${hardcodedTitle}`, () => {
    render(<Header />);
    const el = screen.getByText(hardcodedTitle);
    expect(el).toBeInTheDocument();
  });
  test(`[Given] a title prop [Then] it will render the passed title: ${mockedTitle}`, () => {
    render(<Header title={mockedTitle} />);
    const el = screen.getByText(mockedTitle);
    expect(el).toBeInTheDocument();
  });
  test(`[Given] no subtitle prop [Then] it will render the hardcoded subtitle: ${hardcodedSubtitle}`, () => {
    render(<Header />);
    const el = screen.getByText(hardcodedSubtitle);
    expect(el).toBeInTheDocument();
  });
  test(`[Given] a subtitle prop [Then] it will render the passed subtitle: ${mockedSubtitle}`, () => {
    render(<Header subtitle={mockedSubtitle} />);
    const el = screen.getByText(mockedSubtitle);
    expect(el).toBeInTheDocument();
  });
});
