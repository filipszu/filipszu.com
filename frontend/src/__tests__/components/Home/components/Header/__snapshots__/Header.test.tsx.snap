// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component responsibble for rendering FiliPSZU Software Engineer It will render correctly and match the snapshot. 1`] = `
<div
  className="Logo"
  data-testid="szu-header"
>
  <h1>
    FiliPSZU
  </h1>
  <h4>
    Software Engineer
  </h4>
</div>
`;
